test/store.spec.js

```diff
@@ -93,5 +94,109 @@ describe('Store Tests', function() {
 			done();
 		}).catch(done);
 	});
+
+	it('Store.prototype._filter() should return a Promise rejected', function(done) {
+		store._filter().catch(function(err) {
+			err.message.should.equal('Store.prototype._filter(model, query) is not implemented');
+			done();
+		});
+	});
+
+	it('Store.prototype._get() should return a Promise rejected', function(done) {
+		store._get().catch(function(err) {
+			err.message.should.equal('Store.prototype._get(model, pk) is not implemented');
+			done();
+		});
+	});
+
+	it('Store.prototype._set() should return a Promise rejected', function(done) {
+		store._set().catch(function(err) {
+			err.message.should.equal('Store.prototype._set(model, obj, operation) is not implemented');
+			done();
+		});
+	});
+
+	it('Store.prototype._delete() should return a Promise rejected', function(done) {
+		store._delete().catch(function(err) {
+			err.message.should.equal('Store.prototype._delete(query) is not implemented');
+			done();
+		});
+	});
+
+	describe('Cache Integration should', function() {
+
+		var cache;
+
+		beforeEach(function() {
+			cache = new MyCache();
+		});
+
+		it('have cacheEnabled true', function(done) {
+			var store = new Store();
+			store.cacheEnabled.should.equal(false);
+
+			var store = new Store(cache);
+			store.cacheEnabled.should.equal(true);
+			store.cache.should.equal(cache);
+			done();
+		});
+
+		it('return from cache when get', function(done) {
+			var store = new Store(cache);
+
+			var pk = '1234';
+			var instance = Promise.resolve({ foo: 'bar' });
+			var getStub = sandbox.stub(store, '_get').returns(instance);
+
+			var model = store.define('myModel', {});
+
+			store.get('myModel', pk).then(function(expected) {
+				store.get('myModel', pk).then(function(actual) {
+					(getStub.calledOnce == true).should.equal(true);
+					(getStub.calledWith(model, pk) == true).should.equal(true);
+					actual.should.equal(expected);
+					done();
+				}).catch(done);
+			}).catch(done);
+		});
+
+		it('set to cache when create', function(done) {
+			var obj = { pk: '1234' };
+			var resolved = Promise.resolve(obj);
+
+			var store = new Store(cache);
+			var model = store.define('myModel', { pk: { type: 'String', primaryKey: true } });
+
+			var createStub = sandbox.stub(store, '_set').returns(resolved);
+			var getStub = sandbox.stub(store, '_get').returns(resolved);
+
+			store.create('myModel', obj).then(function(actual) {
+				store.get('myModel', obj.pk).then(function(expected) {
+					(getStub.called === false).should.equal(true);
```

Comment:
[Low] I think it's easier to write it like this `getStub.called.should.equal(false)`